catalog/spatial/kml/spatial-kml-transformer/src/test/java/org/codice/ddf/spatial/kml/transformer/KMLTransformerImplTest.java

```diff
@@ -222,6 +254,71 @@ public void testPerformDefaultTransformationMultiLineStringLocation()
     assertThat(multiLineString.getGeometry().get(1), instanceOf(LineString.class));
   }
 
+  @Test
+  public void testPerformDefaultTransformationExtendedData()
+      throws CatalogTransformerException, DateTimeParseException {
+    MetacardImpl metacard = createMockMetacard();
+
+    metacard.setLocation(POINT_WKT);
+    metacard.setCreatedDate(metacardDate);
+    metacard.setEffectiveDate(metacardDate);
+    metacard.setExpirationDate(metacardDate);
+    metacard.setModifiedDate(metacardDate);
+    metacard.setTags(METACARD_TAGS);
+
+    final Set<AttributeDescriptor> attributeDescriptors =
+        metacard.getMetacardType().getAttributeDescriptors();
+
+    Placemark placemark = kmlTransformer.performDefaultTransformation(metacard);
+
+    final List<Data> dataList = placemark.getExtendedData().getData();
+
+    int dataCount = 0;
+    for (AttributeDescriptor attributeDescriptor : attributeDescriptors) {
+      final String attributeName = attributeDescriptor.getName();
+      final Attribute attribute = metacard.getAttribute(attributeName);
+      if (attribute != null) {
+        dataCount++;
+      }
+    }
+
+    assertThat(dataList.size(), is(dataCount));
+
+    for (Data data : dataList) {
+      switch (data.getName()) {
+        case Core.ID:
+          assertThat(data.getValue(), is(METACARD_ID));
+          break;
+        case Core.TITLE:
+          assertThat(data.getValue(), is(METACARD_TITLE));
+          break;
+        case Core.LOCATION:
+          assertThat(data.getValue(), is(POINT_WKT));
+          break;
+        case Metacard.CONTENT_TYPE:
+          assertThat(data.getValue(), is(METACARD_TYPE));
+          break;
+        case Metacard.CONTENT_TYPE_VERSION:
+          assertThat(data.getValue(), is(METACARD_VERSION));
+          break;
+        case Core.METADATA:
+          assertThat(data.getValue(), is(METACARD_METADATA));
+          break;
+        case Core.METACARD_TAGS:
+          assertThat(
+              data.getValue(),
+              is(METACARD_TAGS.asList().get(0) + "," + METACARD_TAGS.asList().get(1)));
+          break;
+        case Core.MODIFIED:
+        case Metacard.EFFECTIVE:
+        case Core.EXPIRATION:
+        case Core.CREATED:
+          assertThat(data.getValue(), is(METACARD_DATE_STRING));
+          break;
```

Comment:
✏️ Since there's no default case, this test will pass even if there are unexpected entries in the extended data. Consider adding a default case that fails the test.

Reply:
Fixing this will also resolve your Codacy issue.